### PR TITLE
Fix mm pad regresion - more conservative estimation of plannable inputs

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128909

- More conservative estimation of plannable inputs
- Consider constant_pad_nd as pointwise node in concat lowering
- Use aten.cat instead of constant pad ndwhen padding just a single dimension because it can be memory-planned away

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/pad_mm.py [hunk 1]
```diff
@@ -303,6 +303,30 @@ def should_exclude_padding_time(match, arg_name):
     if not fetch_fake_tensors(match, (arg_name,))[0].is_contiguous():
         return False
 
+    # TODO - see issue https://githpub.com/pytorch/pytorch/issues/128889
+    # We would only able to completely plan these out if we were only doing
+    # first dimension padding. non-first we would still need a copy
+    # because these outputs are fixed dense.
+    cannot_plan_output = [
+        aten.mm.default,
```
Comment (Chillee) [inline torch/_inductor/fx_passes/pad_mm.py:311]: This is just because we force the layout with max-autotune today, right?

Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:311]: Yea, i can exclude this when not in max-autotune. Although we'd still need to exclude when padding non first dim

Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:326]: eh, actually, i think it'd be better to be consistent..

**Review by Chillee — APPROVED** [review]

**Pull request closed without merging**